src/d/actor/d_a_ship.cpp

```diff
@@ -5,460 +5,4713 @@
+        current.pos.z = local_b0.z;
+    }
+
+    setYPos();
+
+    float diff = m03F4 - current.pos.y;
+
+    mTrack.mBaseY = m03F4;
+    mTrack.mMinY = m03F8;
+
+    if (!checkStateFlg(daSFLG_UNK1_e)) {
+        setWaveAngle(&local_190, &local_18e);
+    }
+    else {
+        local_190 = m037C;
+        local_18e = m037E;
+    }
+
+    diff += 5.0f;
+
+    if (diff > m03D8) {
+        m03D8 = diff;
+    }
+    else {
+        cLib_chaseF(&m03D8, diff, 3.0f);
+    }
+
+    m037C = shape_angle.x;
+    m037E = shape_angle.z;
+
+    if (m034C == 14) {
+        shape_angle.x = 0;
+        shape_angle.z = 0;
+    }
+    else {
+        shape_angle.x = m0370 + m0384;
+        shape_angle.z = m036C + m0372;
+    }
+
+    mDoMtx_trans(mDoMtx_stack_c::get(), current.pos.x, current.pos.y, current.pos.z);
+    
+    mDoMtx_ZXYrotM(mDoMtx_stack_c::get(), shape_angle.x, shape_angle.y, shape_angle.z);
+    
+    J3DModel* mModel1 = m0298->getModel();
+    J3DModel* mModel2 = m029C->getModel(); 
+    
+    mModel1->setBaseTRMtx(mDoMtx_stack_c::get());
+    m0298->calc();
+    
+    setHeadAnm();
+    
+    mModel2->setBaseTRMtx(mModel1->getAnmMtx(4));
+    m029C->calc();
+
+    if (mShipMode == UsingSalvageArm) {
+        if (m0392 == 10) {
+            m0398 = m039A * getAnglePartRate(); // regswap
+            m039C *= getAnglePartRate();
+            incRopeCnt(-0x14, 0);
+        }
+        mpSalvageArmModel->calc();
+        setRopePos();
+        mDoMtx_multVecZero(mpSalvageArmModel->getAnmMtx(1), &m102C);
+        if (mProc == &daShip_c::procCrane || mProc == &daShip_c::procCraneUp) {
+            m0434 = mRopeLine.mpLines->mpSegments;
+        }
+        else {
+            m0434 = NULL;
+        }
+    }
+    else {
+        m19C0.end();
+        mRipple.end();
+        m19AC.end();
+
+        mCurrentRopeSegmentIndex = 0;
+
+        m0434 = NULL;
+
+        if (mShipMode == UsingCannon) {
+            mDoMtx_multVecZero(mpCannonModel->getAnmMtx(2), &m1038);
+            
+            if (mProc != &daShip_c::procCannon) {
+                m0396 = getAnglePartRate() * 16384.0f;
+                m0394 *= getAnglePartRate();
+            
+            }
+            
+            mpCannonModel->calc();
+            
+            if (m037A == 30) {
+                cMtx_multVec(mpCannonModel->getAnmMtx(2), &l_cannon_top, &cStack_bc);
+
+                local_184.set(getCannonAngleX(), shape_angle.y + m0394, shape_angle.z);
+
+                daBomb_c* this_00 = (daBomb_c *)fopAcM_fastCreate(0x128, daBomb_c::prm_make(daBomb_c::STATE_4, FALSE, TRUE), &cStack_bc, tevStr.mRoomNo, &local_184);
+
+                int bombId;
+                if (this_00) {
+                    if (this_00) {
+                        bombId = ((base_process_class*)this_00)->mBsPcId;
+                    }
+                    else {
+                        bombId = 0xffffffff;
+                    }
+
+                    dCam_getBody()->ForceLockOn(bombId);
```

Comment:
Inline
```suggestion
                    dCam_getBody()->ForceLockOn(fopAcM_GetID(bomb));
```
Also, `this_00` is a bizarre Ghidra-ism, idk why it does that. Just name this bomb since we know its class already

Reply:
Yeah I'm aware of this inline but the asm expects the above code not the inline, it's odd but perhaps the dev didn't use it for whatever reason?

Reply:
That seems unlikely in this particular case since there's a doubled up `if (this_00) {` check that a programmer wouldn't write out twice manually.

If it doesn't match leave a fakematch comment mentioning briefly what issue is caused by the inline

Reply:
I think the issue is because it's in that large execute function, the compiler is doing that thing where it "gives up" and instead of generating the asm it instead just puts the function name
![Screenshot 2025-03-28 at 12 38 43 AM](https://github.com/user-attachments/assets/bff3a389-481f-4312-876d-c844367a4f10)

Reply:
Oh so it's in the same function as the dComIfGp_demo_getActor problem, but it's doing the opposite thing? That's really weird, the number of functions it inlines being wrong could be related to other issues in this function but I don't really get how it would be inlining both too many and too few at the same time, something's up with these inlines